### DISCUSSION TITLE
[IgxGrid] - How to change filter operator per column programmatically?

### DISCUSSION THREAD

**Discussion by mddifilippo89** [IgniteUI/igniteui-angular#12299] [General]
## Question  
How to change filter operator per column programmatically? eg. Great Than, Starts With

eg.
Contains - is default for string columns. 
Equals - is the default for numeric columns

**Comment by ddaribo** [comment]
Hi @mddifilippo89,

If you’d like to modify the default filter operation for a given filtering operand, that is the first one appearing in the list, you could create a custom filtering operand. Sample code and a demo can be found in our documentation [here](https://www.infragistics.com/products/ignite-ui-angular/angular/components/grid/filtering#setup). 

Since your requirement is to essentially perform a reorder of the `operations` array, here is an example logic:
```
export class CustomOrderedStringFilteringOperand extends IgxStringFilteringOperand {
  private constructor() {
    super();

    this.moveStartsWithOperand();
  }
  private moveStartsWithOperand() {
   // 'Starts With' is on index 2 by default
    const startsWithOperand = this.operations[2] as IFilteringOperation;
    this.operations.splice(2, 1);
    this.operations.unshift(startsWithOperand);
  }
}
```
`
 public customOrderedStringFilteringOperand = CustomOrderedStringFilteringOperand.instance();
`

And an object of this class has to be assigned to the target column’s [filters](https://www.infragistics.com/products/ignite-ui-angular/docs/typescript/latest/classes/igxcolumncomponent.html#filters) input:
```
    <igx-column
      field="ProductName"
      header="Product Name"
      [dataType]="'string'"
      [filters]="customOrderedStringFilteringOperand"
    >
```

[Here](https://stackblitz.com/edit/igx-grid-modify-default-fitler-operands?file=src/app/grid/grid-custom-filtering/grid-custom-filtering.component.html) is a small sample demonstrating this. The custom filtering operand and consequently the custom order will be used for all filter modes – quick, excel style and advanced filtering.
